site/content/entry/yuzu-progress-report-mar-2022/index.md

```diff
@@ -0,0 +1,282 @@
+Again, this could have a positive impact on other games, but don’t expect it to be in such a dramatic way.
+
+[Merry](https://github.com/merryhime) updated [dynarmic](https://github.com/merryhime/dynarmic) to the latest version to provide better support for `Super Mario Galaxy`, {{< gh-hovercard "8054" "adding support for accelerated SHA256 CPU instructions" >}}, among other changes.
+This avoids softlocking Galaxy at the final level.
+
+{{< single-title-imgs
+	"This toad sure likes to talk a lot (Super Mario Galaxy)"
+	"./creditbug.mp4"
+	"./creditfix.mp4"
+>}}
+
+## POYO!
+
+`Kirby and the Forgotten Land` turned out to be one of the best platformers of recent times, and also quite a challenge for our devs.
+
+epicboy jumped in to keep the ball rolling and started by solving the geometry pop-ins that are very noticeable in the first gameplay scenes.
+
+{{< single-title-imgs
+	"That tree just pops! (Kirby and the Forgotten Land)"
+	"./popbug.mp4"
+	"./popfix.mp4"
+>}}
+
+This was achieved by {{< gh-hovercard "8023" "improving the accuracy of the constant buffer uploads" >}}, and has the added benefit of also fixing rendering issues in `Monster Hunter Ries` and `Diablo® II: Resurrected™`.
+
+{{< imgs
+	"./rise.png| Such a pretty intro (MONSTER HUNTER RISE)"
+  >}}
+
+&nbsp;
+
+{{< imgs
+	"./diablo2.png| A real classic (Diablo® II: Resurrected™)"
+  >}}
+
+As you can see, Diablo II has some UI rendering issues, but at least every character has access to all their limbs now!
+
+Another issue that was quickly pointed out, is that Nvidia hardware would eat VRAM like crazy, to the point of filling up even high end hardware in minutes.
+
+{{< imgs
+	"./kirbyvram.png| PC requirements: RTX 3090? Not on our watch!"
+  >}}
+
+{{< gh-hovercard "8093" "Unmapping pre-existing rasterizer memory before the memory range was mapped" >}} solves the issue and allows Nvidia users to not require RTX 3090 levels of hardware to get a good experience.
+
+epicboy also {{< gh-hovercard "8116" "fixed an svc break crash if the game was run with Asynchronous GPU Emulation was enabled" >}}, an option that should always remain enabled.
+
+And finally, an off-by-one error in the stream buffer was responsible for causing vertex explosions.
+The solution is to {{< gh-hovercard "8127" "simplify the implementation of stream buffers" >}}.
+As the new implementation can only feed a single upload request at a time, this may provide a small performance loss in some cases, but it’s sure to fix several other games in the process.
+
+{{< single-title-imgs-compare
+	"A look inside Kirby's powers? (Kirby and the Forgotten Land)"
+	"./poyobug.png"
+	"./poyofix.png"
+>}}
+
+AMD Radeon users will suffer from more stuttering than usual when running this game.
+This is caused because neither the Windows AMD drivers nor any of the Linux options offer support for the `VK_EXT_vertex_input_dynamic_state` extension, which helps reduce the shader count by 3 times the original amount.
```

Comment:
RADV does support the extension, but it's blacklisted on RDNA2 hardware bc it causes crashes there. (See last month's progress report ;) )